File: android/src/main/java/expo/modules/brightnesscontroller/BrightnessControllerModule.kt
package expo.modules.brightnesscontroller

import android.content.Context
import android.view.WindowManager
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class BrightnessControllerModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("BrightnessController")

        Function("setBrightness") { brightness: Float ->
            val activity = appContext.currentActivity
            if (activity != null) {
                val window = activity.window
                val layoutParams = window.attributes
                layoutParams.screenBrightness = brightness
                window.attributes = layoutParams
            }
        }
    }
}

File: ios/BrightnessControllerModule.swift
// import ExpoModulesCore

// public class BrightnessControllerModule: Module {
//     public func definition() -> ModuleDefinition {
//         Name("BrightnessController")

//         Function("setBrightness") { (brightness: Float) in
//             UIScreen.main.brightness = CGFloat(brightness)
//         }
//     }
// }

import ExpoModulesCore

public class BrightnessControllerModule: Module {
    public func definition() -> ModuleDefinition {
        Name("BrightnessController")

        AsyncFunction("setBrightness") { (brightness: Float, promise: Promise) in
            DispatchQueue.main.async {
                UIScreen.main.brightness = CGFloat(brightness)
                promise.resolve(nil)
            }
        }
    }
}
